SavedInstances/Modules/Progress.lua

```diff
@@ -368,6 +368,38 @@ local function PrimalStormsCoreReset(toon, index)
   wipe(t.Progress[index])
 end
 
+local function SparksOfLifeUpdate(index)
+  SI.db.Toons[SI.thisToon].Progress[index] = wipe(SI.db.Toons[SI.thisToon].Progress[index] or {})
+  for _, questID in ipairs(Module.TrackedQuest[index].relatedQuest) do
+    SI.db.Toons[SI.thisToon].Progress[index][questID] = C_TaskQuest_IsActive(questID)
+    -- Is active
+    if SI.db.Toons[SI.thisToon].Progress[index][questID] then
+      local result = {}
+      local _, objectiveType, finished, numFulfilled, numRequired = GetQuestObjectiveInfo(questID, 1, false)
+      result.objectiveType = objectiveType
+      result.isFinish = finished
+      result.numFulfilled = numFulfilled
+      result.numRequired = numRequired
+      if C_QuestLog_IsQuestFlaggedCompleted(questID) then
+        result.unlocked = true
+        result.isComplete = true
+      else
+        local isOnQuest = C_QuestLog_IsOnQuest(questID)
+        result.unlocked = isOnQuest
+        result.isComplete = false
+      end
+      SI.db.Toons[SI.thisToon].Progress[index] = result
+    end
```

Comment:
Some problems in these code. The quest active status is assigned in line 374 but the whole table is replaced in line 391 to wipe out the previous store.

As only one quest will be active in one time. Don't store the quest that is active in database and reuse the table wiped previously. Just do

```lua
if C_TaskQuest_IsActive(questID) then
  local result = SI.db.Toons[SI.thisToon].Progress[index]
  local _, objectiveType, finished, numFulfilled, numRequired = GetQuestObjectiveInfo(questID, 1, false)
```

And change line 391 to `break`.

Reply:
Ah yes I see. Updated!